components/expense/expense_api.go

```diff
@@ -0,0 +1,99 @@
+package expense
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/marian0/expenses-go-api/common"
+)
+
+//API -
+type API struct {
+	eService    *Service
+}
+
+//NewAPI -
+func NewAPI(expenseService *Service) *API {
+
+
+	return &API{
+		eService:	expenseService,
+	}
+}
+
+//FindAll -
+func (eAPI *API) FindAll(c *gin.Context) {
+	expenses := eAPI.eService.FindAll()
+
+	common.RespondSuccess(c, http.StatusOK, gin.H{"expenses": ToExpenseDTOs(expenses)})
+}
+
+//FindByID -
+func (eAPI *API) FindByID(c *gin.Context) {
+	id, _ := strconv.Atoi(c.Param("id"))
```

Comment:
Please, add error checking and respond a validation failed here : 

```
if err != nil {
    common.RespondError(c, http.StatusBadRequest, err.Error(), "There was a problem trying to retrieve Expense")
}
```

As well as checking if no expense found, to return a 404 error

Reply:
something like this 👇 

```
	if expense == (Expense{}) {
		common.RespondError(c, http.StatusNotFound, "Expense not found "+c.Param("id"), "Expense not found")
		return
	}

```